lib/resty/core/shdict.lua

```diff
@@ -387,6 +401,25 @@ local function shdict_flush_all(zone)
 end
 
 
+local function shdict_get_used_bytes(zone)
+    zone = check_zone(zone)
+
+    local slot_count_buf = ffi_new("size_t[1]")
+    C.ngx_http_lua_ffi_shdict_get_stats_slot_count(zone, slot_count_buf)
+    local stats = ffi_new("ngx_slab_stat_t[?]", slot_count_buf[0])
+    local min_size = ffi_new("size_t[1]")
+    C.ngx_http_lua_ffi_shdict_get_stats_slots_and_min_size(zone, slot_count_buf[0], stats, min_size)
+    local slot_count = tonumber(slot_count_buf[0])
+    local slot_size = tonumber(min_size[0])
+    local total_used = 0
+    for i = 1, slot_count do
+        total_used = total_used + slot_size * tonumber(stats[i-1].used)
+        slot_size = slot_size * 2
+    end
+    return total_used
```

Comment:
It might be useful to have the more detailed stats in each slot. Not just the totals.